src/components/Widgets/variants/base/ZapWidget/ZapDepositBackendWidget.tsx

```diff
@@ -0,0 +1,93 @@
+'use client';
+
+import { HiddenUI, LiFiWidget } from '@lifi/widget';
+import { FC, useMemo } from 'react';
+import { useEnhancedZapData } from 'src/hooks/zaps/useEnhancedZapData';
+import { useZapQuestIdStorage } from 'src/providers/hooks';
+import { useWidgetTrackingContext } from 'src/providers/WidgetTrackingProvider';
+import { useMenuStore } from 'src/stores/menu/MenuStore';
+import { useLiFiWidgetConfig } from '../../widgetConfig/hooks';
+import { ConfigContext } from '../../widgetConfig/types';
+import { WidgetProps } from '../Widget.types';
+import { ZapDepositSettings } from './ZapDepositSettings';
+
+interface ZapDepositBackendWidgetProps extends WidgetProps {}
+
+export const ZapDepositBackendWidget: FC<ZapDepositBackendWidgetProps> = ({
+  customInformation,
+  ctx,
+}) => {
+  useZapQuestIdStorage();
+
+  const projectData = useMemo(() => {
+    return customInformation?.projectData;
+  }, [customInformation?.projectData]);
+
+  const {
+    zapData,
+    isSuccess: isZapDataSuccess,
+    depositTokenData,
+    depositTokenDecimals,
+    isLoadingDepositTokenData,
+    refetchDepositToken,
+  } = useEnhancedZapData(projectData);
+
+  const { setDestinationChainTokenForTracking } = useWidgetTrackingContext();
+
+  const [setSupportModalState] = useMenuStore((state) => [
+    state.setSupportModalState,
+  ]);
+
+  // const poolName = useMemo(() => {
+  //   return `${zapData?.meta.name} ${zapData?.market?.depositToken?.symbol.toUpperCase()} Pool`;
+  // }, [JSON.stringify(zapData ?? {})]);
+
+  const toToken = useMemo(() => {
+    return zapData?.market?.depositToken.address;
+  }, [JSON.stringify(zapData ?? {})]);
+
+  const toChain = useMemo(() => {
+    return zapData?.market?.depositToken.chainId;
+  }, [JSON.stringify(zapData ?? {})]);
```

Comment:
What's the point of these useMemo?

this code seems simpler, faster, and won't change the number of rerenders:

```suggestion
  const toToken = zapData?.market?.depositToken.address
  const toChain = zapData?.market?.depositToken.chainId
```

Reply:
was already there